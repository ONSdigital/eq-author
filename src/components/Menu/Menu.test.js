import React from "react";
import { mount } from "enzyme";

import {
  Dropdown,
  MenuItem,
  MenuList,
  SubMenuItem,
  MenuScrollPane
} from "components/Menu";

let handleItemChosen, wrapper, item;

describe("components/Menu", () => {
  beforeEach(() => {
    handleItemChosen = jest.fn();
    item = { id: "123" };

    wrapper = mount(
      <Dropdown>
        <MenuScrollPane>
          <MenuList>
            <MenuItem id="item-1" item={item} onItemChosen={handleItemChosen}>
              1. Fusce Egestas Tristique Porta Sollicitudin
            </MenuItem>
            <MenuItem id="item-2" item={item} onItemChosen={handleItemChosen}>
              2. Fusce Egestas Tristique Porta Sollicitudin
            </MenuItem>
            <MenuItem id="item-3" item={item} onItemChosen={handleItemChosen}>
              3. Fusce Egestas Tristique Porta Sollicitudin
            </MenuItem>
            <SubMenuItem
              menu={
                <Dropdown>
                  <MenuList>
                    <MenuItem
                      id="item-4.1"
                      item={item}
                      onItemChosen={handleItemChosen}
                    >
                      4.1. Fusce Egestas Tristique Porta Sollicitudin
                    </MenuItem>
                    <MenuItem
                      id="item-4.2"
                      item={item}
                      onItemChosen={handleItemChosen}
                    >
                      4.2. Ligula Consectetur Ullamcorper Cursus Ligula
                      Consectetur Ullamcorper Cursus
                    </MenuItem>
                  </MenuList>
                </Dropdown>
              }
            >
              4. Fusce Egestas Tristique Porta Sollicitudin
            </SubMenuItem>
          </MenuList>
        </MenuScrollPane>
      </Dropdown>
    );
  });

  it("should render a Menu", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle items being chosen by calling onItemChosen with id prop", () => {
    const menuItem = wrapper.find(MenuItem).first();
    menuItem.simulate("click");
    expect(handleItemChosen).toHaveBeenCalledWith(item);
  });
});
