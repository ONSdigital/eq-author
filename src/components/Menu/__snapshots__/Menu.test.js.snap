// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Menu should render a Menu 1`] = `
.c0 {
  background: #FFFFFF;
  border-top: 1px solid rgba(0,0,0,0.04);
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.24);
  border-radius: 0 2px 2px 0;
}

.c4 {
  margin: 0.5em 0;
  overflow: hidden;
}

.c3 {
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  line-height: 1.3;
  font-size: 0.8em;
  padding: 0 1em;
}

.c3.is-highlighted {
  background: #f5f5f5;
}

.c5 {
  overflow: hidden;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  line-height: 1.3;
  font-size: 0.8em;
  padding: 0 1em;
  padding-right: 2em;
}

.c5.is-highlighted {
  background: #f5f5f5;
}

.c5::after {
  content: "";
  opacity: 0.5;
  display: block;
  background: transparent url(icon-arrow.svg) no-repeat center;
  position: absolute;
  right: 0.5em;
  top: 0;
  bottom: 0;
  margin: auto;
  height: 22px;
  width: 12px;
}

.c5::before {
  content: "";
  width: 7px;
  display: block;
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  margin: auto;
  height: 100%;
}

.c2 {
  width: 100%;
  height: 100%;
  overflow-y: auto;
  position: relative;
  -webkit-background-clip: text;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  background-color: rgba(0,0,0,0.18);
}

.c2::-webkit-scrollbar-thumb {
  border-radius: 0;
  box-shadow: none;
  background-color: #c3c3c3;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c2::-webkit-scrollbar-track {
  border-radius: 0;
  box-shadow: none;
  display: none;
}

.c2::-webkit-scrollbar {
  width: 10px;
}

.c2 >:first-child {
  -webkit-backface-visibility: hidden !important;
  backface-visibility: hidden !important;
}

.c2:hover::-webkit-scrollbar-thumb {
  background-color: #aaa;
}

.c1 {
  overflow-y: auto;
  overflow-x: hidden;
}

.c1:hover {
  background-color: rgba(0,0,0,0.18);
}

.c1:hover::-webkit-scrollbar-thumb {
  background: rgba(0,0,0,0.2);
}

.c1::-webkit-scrollbar {
  width: 7px;
}

<Dropdown>
  <div
    className="c0"
  >
    <MenuScrollPane>
      <ScrollPane
        className="c1"
        permanentScrollBar={false}
      >
        <div
          className="c1 c2"
        >
          <MenuList>
            <MenuScrollPane>
              <ScrollPane
                className="c1"
                permanentScrollBar={false}
              >
                <div
                  className="c1 c2"
                >
                  <MenuList>
                    <div
                      role="menu"
                    >
                      <MenuItem
                        id="item-1"
                        item={
                          Object {
                            "id": "123",
                          }
                        }
                        lines={1}
                        onItemChosen={[Function]}
                      >
                        <MenuItem__StyledMenuItem
                          highlightedClassName="is-highlighted"
                          id="item-1"
                          lines={1}
                          onItemChosen={[Function]}
                        >
                          <MenuItem
                            className="c3"
                            highlightedClassName="is-highlighted"
                            id="item-1"
                            lines={1}
                            onItemChosen={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              role="menuitem"
                            >
                              <MenuItem__MenuItemInner>
                                <div
                                  className="c4"
                                >
                                  1. Fusce Egestas Tristique Porta Sollicitudin
                                </div>
                              </MenuItem__MenuItemInner>
                            </div>
                          </MenuItem>
                        </MenuItem__StyledMenuItem>
                      </MenuItem>
                      <MenuItem
                        id="item-2"
                        item={
                          Object {
                            "id": "123",
                          }
                        }
                        lines={1}
                        onItemChosen={[Function]}
                      >
                        <MenuItem__StyledMenuItem
                          highlightedClassName="is-highlighted"
                          id="item-2"
                          lines={1}
                          onItemChosen={[Function]}
                        >
                          <MenuItem
                            className="c3"
                            highlightedClassName="is-highlighted"
                            id="item-2"
                            lines={1}
                            onItemChosen={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              role="menuitem"
                            >
                              <MenuItem__MenuItemInner>
                                <div
                                  className="c4"
                                >
                                  2. Fusce Egestas Tristique Porta Sollicitudin
                                </div>
                              </MenuItem__MenuItemInner>
                            </div>
                          </MenuItem>
                        </MenuItem__StyledMenuItem>
                      </MenuItem>
                      <MenuItem
                        id="item-3"
                        item={
                          Object {
                            "id": "123",
                          }
                        }
                        lines={1}
                        onItemChosen={[Function]}
                      >
                        <MenuItem__StyledMenuItem
                          highlightedClassName="is-highlighted"
                          id="item-3"
                          lines={1}
                          onItemChosen={[Function]}
                        >
                          <MenuItem
                            className="c3"
                            highlightedClassName="is-highlighted"
                            id="item-3"
                            lines={1}
                            onItemChosen={[Function]}
                          >
                            <div
                              className="c3"
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              role="menuitem"
                            >
                              <MenuItem__MenuItemInner>
                                <div
                                  className="c4"
                                >
                                  3. Fusce Egestas Tristique Porta Sollicitudin
                                </div>
                              </MenuItem__MenuItemInner>
                            </div>
                          </MenuItem>
                        </MenuItem__StyledMenuItem>
                      </MenuItem>
                      <SubMenuItem
                        lines={1}
                        menu={
                          <Dropdown>
                            <MenuList>
                              <MenuItem
                                id="item-4.1"
                                item={
                                  Object {
                                    "id": "123",
                                  }
                                }
                                lines={1}
                                onItemChosen={[Function]}
                              >
                                4.1. Fusce Egestas Tristique Porta Sollicitudin
                              </MenuItem>
                              <MenuItem
                                id="item-4.2"
                                item={
                                  Object {
                                    "id": "123",
                                  }
                                }
                                lines={1}
                                onItemChosen={[Function]}
                              >
                                4.2. Ligula Consectetur Ullamcorper Cursus Ligula Consectetur Ullamcorper Cursus
                              </MenuItem>
                            </MenuList>
                          </Dropdown>
                        }
                      >
                        <SubMenuItem__StyledSubMenuItem
                          highlightedClassName="is-highlighted"
                          lines={1}
                          menu={
                            <Dropdown>
                              <MenuList>
                                <MenuItem
                                  id="item-4.1"
                                  item={
                                    Object {
                                      "id": "123",
                                    }
                                  }
                                  lines={1}
                                  onItemChosen={[Function]}
                                >
                                  4.1. Fusce Egestas Tristique Porta Sollicitudin
                                </MenuItem>
                                <MenuItem
                                  id="item-4.2"
                                  item={
                                    Object {
                                      "id": "123",
                                    }
                                  }
                                  lines={1}
                                  onItemChosen={[Function]}
                                >
                                  4.2. Ligula Consectetur Ullamcorper Cursus Ligula Consectetur Ullamcorper Cursus
                                </MenuItem>
                              </MenuList>
                            </Dropdown>
                          }
                          positionOptions={
                            Object {
                              "hAlign": Array [
                                "center",
                                "left",
                                "right",
                              ],
                              "position": "right",
                              "vAlign": Array [
                                "top",
                                "center",
                                "bottom",
                              ],
                            }
                          }
                        >
                          <SubMenuItem
                            className="c5"
                            highlightedClassName="is-highlighted"
                            lines={1}
                            menu={
                              <Dropdown>
                                <MenuList>
                                  <MenuItem
                                    id="item-4.1"
                                    item={
                                      Object {
                                        "id": "123",
                                      }
                                    }
                                    lines={1}
                                    onItemChosen={[Function]}
                                  >
                                    4.1. Fusce Egestas Tristique Porta Sollicitudin
                                  </MenuItem>
                                  <MenuItem
                                    id="item-4.2"
                                    item={
                                      Object {
                                        "id": "123",
                                      }
                                    }
                                    lines={1}
                                    onItemChosen={[Function]}
                                  >
                                    4.2. Ligula Consectetur Ullamcorper Cursus Ligula Consectetur Ullamcorper Cursus
                                  </MenuItem>
                                </MenuList>
                              </Dropdown>
                            }
                            positionOptions={
                              Object {
                                "hAlign": Array [
                                  "center",
                                  "left",
                                  "right",
                                ],
                                "position": "right",
                                "vAlign": Array [
                                  "top",
                                  "center",
                                  "bottom",
                                ],
                              }
                            }
                          >
                            <FloatAnchor
                              anchor={
                                <MenuItem
                                  aria-expanded={false}
                                  aria-haspopup={true}
                                  className="c5"
                                  highlightedClassName="is-highlighted"
                                  highlightedStyle={undefined}
                                  index={undefined}
                                  onHighlightChange={[Function]}
                                  onItemChosen={[Function]}
                                  onMouseLeave={[Function]}
                                  onRightPushed={[Function]}
                                  style={undefined}
                                >
                                  <MenuItem__MenuItemInner>
                                    4. Fusce Egestas Tristique Porta Sollicitudin
                                  </MenuItem__MenuItemInner>
                                </MenuItem>
                              }
                              float={null}
                              options={
                                Object {
                                  "hAlign": Array [
                                    "center",
                                    "left",
                                    "right",
                                  ],
                                  "position": "right",
                                  "vAlign": Array [
                                    "top",
                                    "center",
                                    "bottom",
                                  ],
                                }
                              }
                            >
                              <MenuItem
                                aria-expanded={false}
                                aria-haspopup={true}
                                className="c5"
                                highlightedClassName="is-highlighted"
                                onHighlightChange={[Function]}
                                onItemChosen={[Function]}
                                onMouseLeave={[Function]}
                                onRightPushed={[Function]}
                              >
                                <div
                                  aria-expanded={false}
                                  aria-haspopup={true}
                                  className="c5"
                                  onClick={[Function]}
                                  onMouseEnter={[Function]}
                                  onMouseLeave={[Function]}
                                  role="menuitem"
                                >
                                  <MenuItem__MenuItemInner>
                                    <div
                                      className="c4"
                                    >
                                      4. Fusce Egestas Tristique Porta Sollicitudin
                                    </div>
                                  </MenuItem__MenuItemInner>
                                </div>
                              </MenuItem>
                            </FloatAnchor>
                          </SubMenuItem>
                        </SubMenuItem__StyledSubMenuItem>
                      </SubMenuItem>
                    </div>
                  </MenuList>
                </div>
              </ScrollPane>
            </MenuScrollPane>
          </MenuList>
        </div>
      </ScrollPane>
    </MenuScrollPane>
  </div>
</Dropdown>
`;
